easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,191 @@
+            regex_subs.append((r"(?P<var>MKLML_ROOT=).*$", r"# \g<var>%s" % os.environ["MKLROOT"]))
+        elif toolchain_blas in ['ACML', 'ATLAS']:
+            blas = "atlas"
+        elif toolchain_blas == 'OpenBLAS':
+            blas = "openblas"
+        elif toolchain_blas is None:
+            # This toolchain has no BLAS library
+            raise EasyBuildError("No BLAS library found in the toolchain")
+
+        regex_subs.append((r'USE_BLAS =.*', 'USE_BLAS = %s' % blas))
+
+        if get_software_root('NNPACK'):
+            regex_subs.append(("USE_NNPACK = 0", "USE_NNPACK = 1"))
+
+        apply_regex_substitutions('config.mk', regex_subs)
+
+        super(EB_MXNet, self).configure_step()
+
+    def install_step(self):
+        """Specify list of files to copy"""
+        self.cfg['files_to_copy'] = ['bin', 'include', 'lib',
+                                     (['dmlc-core/include/dmlc', 'nnvm/include/nnvm'], 'include')]
+        super(EB_MXNet, self).install_step()
+
+    def extensions_step(self):
+        """Build & Install both Python and R extension"""
+        # we start with the python bindings
+        self.py_ext = PythonPackage(self, {'name': 'mxnet'})
+        self.py_ext.module_generator = self.module_generator
+        self.py_ext.src = os.path.join(self.mxnet_src_dir, "python")
+        os.chdir(self.py_ext.src)
+
+        self.py_ext.prerun()
+        self.py_ext.run(unpack_src=False)
+        self.py_ext.postrun()
+
+        # next up, the R bindings
+        self.r_ext = RPackage(self, {'name': 'R-package'})
+        self.r_ext.module_generator = self.module_generator
+        self.r_ext.src = os.path.join(self.mxnet_src_dir, "R-package")
+        try:
+            os.chdir(self.r_ext.src)
+            mkdir("inst")
+            symlink(os.path.join(self.installdir, "lib"), "inst/libs")
+            symlink(os.path.join(self.installdir, "include"), "inst/include")
+        except IOError, err:
+            raise EasyBuildError("Failed to prepare the 'inst' directory for the R bindings: %s", err)
+
+        namespace = ["import(Rcpp)", "import(methods)", ""]
+        write_file("NAMESPACE", "\n".join(namespace))
+        self.r_ext.prerun()
+        self.r_ext.run()
+        self.r_ext.postrun()
+
+    def sanity_check_step(self):
+        """Check for main library files for MXNet"""
+        custom_paths = {
+            'files': ["lib/libmxnet.%s" % ext for ext in ['a', get_shared_lib_ext()]],
+            'dirs': [],
+        }
+        super(EB_MXNet, self).sanity_check_step(custom_paths=custom_paths)
+        self.py_ext.sanity_check_step()
+        self.r_ext.sanity_check_step()
+
+    def make_module_extra(self, *args, **kwargs):
+        """Custom variables for MXNet module."""
+        txt = super(EB_MXNet, self).make_module_extra(*args, **kwargs)
+        txt += self.py_ext.make_module_extra(*args, **kwargs)
+        txt += self.r_ext.make_module_extra(*args, **kwargs)
+        return txt
```

Comment:
this gives:
```tcl
prepend-path	CPATH		$root/include
prepend-path	LD_LIBRARY_PATH		$root/lib
prepend-path	LIBRARY_PATH		$root/lib
prepend-path	PATH		$root/bin
setenv	EBROOTMXNET		"$root"
setenv	EBVERSIONMXNET		"0.9.3"
setenv	EBDEVELMXNET		"$root/easybuild/MXNet-0.9.3-foss-2016b-Python-2.7.12-easybuild-devel"

setenv	EBROOTMXNET		"$root"
setenv	EBVERSIONMXNET		None
setenv	EBDEVELMXNET		"$root/easybuild/mxnet-None-foss-2016b-Python-2.7.12-easybuild-devel"

prepend-path	PYTHONPATH		$root/lib/python2.7/site-packages
setenv	EBROOTMXNET		"$root"
setenv	EBVERSIONMXNET		None
setenv	EBDEVELMXNET		"$root/easybuild/mxnet-None-foss-2016b-Python-2.7.12-easybuild-devel"

prepend-path	R_LIBS		$root
```

Any way to use the logic from the extensions easyblock to figure out the `PYTHONPATH` and `R_LIBS` without have duplicate entries?